dynamo/preprocessing/preprocessor_utils.py

```diff
@@ -1263,24 +1273,20 @@ def calc_sz_factor(
                 _adata = _adata[:, _adata.var_names.intersection(genes_use_for_norm)]
 
     if total_layers is not None:
-        if not isinstance(total_layers, list):
-            total_layers = [total_layers]
-        if len(set(total_layers).difference(_adata.layers.keys())) == 0:
-            total = None
-            for t_key in total_layers:
-                total = _adata.layers[t_key] if total is None else total + _adata.layers[t_key]
-            _adata.layers["_total_"] = total
-            layers.extend(["_total_"])
+        total_layers, layers = DKM.aggregate_layers_into_total(
+            _adata,
+            layers=layers,
+            total_layers=total_layers,
+        )
 
     layers = DKM.get_available_layer_keys(_adata, layers)
     if "raw" in layers and _adata.raw is None:
         _adata.raw = _adata.copy()
 
-    excluded_layers = []
-    if not X_total_layers:
-        excluded_layers.extend(["X"])
-    if not splicing_total_layers:
-        excluded_layers.extend(["spliced", "unspliced"])
+    excluded_layers = DKM.get_excluded_layers(
```

Comment:
add some annotation to make it clear that, when splicing_total_layers is False, we normalize spliced and unspliced RNA separately using each layer's size factors. Similary, when  X_total_layers is False, we normalize X (normally it corresponds to the spliced RNA or total RNA for a conventional scRNA-seq or labeling scRNA-seq) using its own's size factor